tests/base.py

```diff
@@ -334,6 +334,7 @@ def modify_expected_record(self, expected_record):
                     self.sort_array_type(expected_record, key, value)
                 else:
                     self.align_date_type(expected_record, key, value)
+                    self.align_number_type(expected_record, key, value)
```

Comment:
I feel like this method is inconsistent. It goes if type is dict, do something, list do something, but then else and then each of the methods checks the type again. Why not put the type checking inside the modify part and then you don't have to do the type checking inside the methods?

Reply:
This whole method needs to be refactored, the recursion was implemented with sorting records in mind which is not a valid modifier. We can address this in a future PR. I will make a ticket to clean up all tests and include this point so that this does not get forgotten.